ui-v2/src/components/ui/stepper.stories.tsx

```diff
@@ -0,0 +1,57 @@
+import { useStepper } from "@/hooks/use-stepper";
+import type { Meta, StoryObj } from "@storybook/react";
+import { Button } from "./button";
+import { Step, type StepState, Stepper } from "./stepper";
+
+const meta: Meta<typeof Stepper> = {
+	title: "UI/Stepper",
+	component: Stepper,
+	render: () => <StepperStory />,
+};
+export default meta;
+
+export const story: StoryObj = { name: "Stepper" };
+
+const STEPS = ["Trigger", "Actions", "Details"] as const;
+
+const StepperStory = () => {
+	const stepper = useStepper(STEPS.length);
+
+	return (
+		<div className="flex flex-col gap-4">
+			<Stepper>
+				{STEPS.map((step, i) => {
+					let state: StepState = "todo";
+					if (stepper.getIsCurrentStep(i)) {
+						state = "active";
+					}
+					if (stepper.getIsStepCompleted(i)) {
+						state = "done";
+					}
+					return (
+						<Step
+							key={step}
+							disabled={stepper.currentStep < i}
+							onClick={(step) => stepper.setCurrentStep(step.number)}
+							number={i}
+							state={state}
+							name={step}
+						/>
+					);
+				})}
+			</Stepper>
+			<div className="flex justify-between">
+				<Button
+					variant="secondary"
+					disabled={stepper.isStartingStep}
+					onClick={stepper.decrementStep}
+				>
+					Previous
+				</Button>
+				<Button disabled={stepper.isFinalStep} onClick={stepper.incrementStep}>
+					Next
+				</Button>
+			</div>
+		</div>
+	);
+};
```

Comment:
I agree, let met try it out

Reply:
alright, updated the component to be more encapsulated.

You should only need something like this (`useStepper` state still needs to be controlled in the parent container)
```
	const stepper = useStepper(STEPS.length);
	return (<Stepper
			steps={STEPS}
			currentStepNum={stepper.currentStep}
			onClick={(step) => stepper.changeStep(step.stepNum)}
			completedSteps={stepper.completedStepsSet}
			visitedSteps={stepper.visitedStepsSet} />)
```

Need to update and cleanup functions in `useStepper`

https://github.com/user-attachments/assets/c65cf2c9-37ba-444b-9434-9e3e89c2f160

Reply:
I was thinking of even moving the `useStepper` and forward and backward buttons into the `Stepper`, but now I'm waffling on what the optimal level of abstraction is. Since `Stepper` will often be used with forms on each step, could you write a story with forms on each step? I think seeing that will help expose the right interface for this.

Reply:
![Screenshot 2024-12-31 at 10 03 40 AM](https://github.com/user-attachments/assets/0379ca15-10d0-48f5-98b4-a49db7dfefc1)

My idea was to abstract the `forward` and `backward` buttons outside of the Stepper component, because when we add a `Form component` and `form state`, ideally the `forward` button should be dependent on the `form state` **and** `useStepper` state.

I was thinking of having the page layout be something like
```
const App = () => {
	const stepper = useStepper();
	const formState = useForm();
	return (
		<div className="...">
			<Stepper {...stepper} />

			<Card>
				<Form>
					<Input />
					<Input />
				</Form>
			</Card>

			<div>
				<Button>Previous</Button>
				<Button disabled={stepper.isLastStep || formState.isDisabled}>
					Next
				</Button>
			</div>
		</div>
	);
};
```
Overall, the challenge I'm seeing is that the `Previous` and `Next` buttons are dependent on both the stepper state **and** form state.

But I'll add a story of what an example form step might be as a story

Reply:
ok, latest push incorporates a story that uses Forms and incremental steps.
It does make it more intuitive to not include the `Previous` and `Next` buttons because we need to tie form validation logic to them


https://github.com/user-attachments/assets/024979cc-b090-4047-ba0b-4ef0db60dc64

Reply:
Cool, that makes the interaction much clearer! Thanks!